node/client.go

```diff
@@ -405,6 +417,153 @@ class Core {
       }
     },
   };
+
+  auth = {
+    // Manually set the session with an access and refresh token acquired elsewhere.
+    // We recommend to rather use the built-in authentication functions provided in this client.
+    setSession: (accessToken: string, refreshToken: string) => {
+      this.session = { accessToken, refreshToken };
+    },
+
+    // Retrieves the current session's access and refresh tokens. This will also refresh the 
+    // session with the authentication server.
+    getSession: async () => {
+      await this.auth.refreshSession();
+      return this.session;
+    },
+
+    // Returns the list of supported auth providers and their SSO login URLs.
+    providers: async (): Promise<Provider[]> => {
+      const authUrl = this.config.baseUrl.replace("/api", "");
+
+      const result = await globalThis.fetch(
+        stripTrailingSlash(authUrl) + "/auth/providers",
+        {
+          method: "GET",
+          cache: "no-cache",
+          headers: {
+            "content-type": "application/json",
+          },
+        },
+      );
+
+      if (result.status == 200) {
+        const rawJson = await result.text();
+        return JSON.parse(rawJson);
+      } else {
+        throw new Error("unexpected status code response from /auth/providers: " + result.status)
+      }
+    },
+
+    // Authenticate with an OIDC token.
+    authenticateWithIdToken: async (idToken: string) => {
+      const req: TokenExchangeGrant = {
+        grant: "token_exchange",
+        subjectToken: idToken
+      }
+
+      await this.auth.tokenRequest(req)
+    },
+
+    // Authenticate with an SSO Login code.
+    authenticateWithSSOLoginCode: async (code: string) => {
+      const req: AuthorizationCodeGrant = {
+        grant: "authorization_code",
+        code: code
+      }
+
+      await this.auth.tokenRequest(req)
+    },
+
+    // Refresh the session with the authentication server.
+    refreshSession: async () => {
+      if (!this.session) {
+        return;
+      }
+
+      const req: RefreshGrant = {
+        grant:  "refresh_token",
+        refreshToken: this.session!.refreshToken
+      }
+
+      await this.auth.tokenRequest(req);
+    },
+
+    tokenRequest: async(req: TokenGrant) => {
+      let body = null;      
+      switch (req.grant) {
+        case "token_exchange":
+          body = {
+            "grant_type": req.grant,
+            "subject_token": req.subjectToken
+          };
+          break;
+        case "authorization_code":
+          body = {
+            "grant_type": req.grant,
+            "code": req.code
+          };
+          break;
+        case "refresh_token":
+          body = {
+            "grant_type": req.grant,
+            "refresh_token": req.refreshToken
+          };
+          break;
+        default:
+          throw new Error("grant not implemented")
```

Comment:
We can do better here.

```suggestion
          throw new Error("Unknown grant type. We currently support `authorization_code`, `token_exchange`, and `refresh_token` grant types. Please use one of those. For more info, please refer to the docs at https://docs.keel.so/authentication/endpoints#parameters")
```